db/compaction/compaction_job.cc

```diff
@@ -1449,8 +1449,9 @@ Status CompactionJob::FinishCompactionOutputFile(
   } else {
     sub_compact->builder->Abandon();
   }
+  IOStatus io_s = sub_compact->builder->io_status();
   if (s.ok()) {
```

Comment:
I find it kind of confusing that this same conditional appears not far below, although it seems correct - https://github.com/facebook/rocksdb/blob/ad10d67b2853d3d31f6bc083c7c5a097363148d4/db/compaction/compaction_job.cc#L1480-L1482

It also feels confusing there's a comment later saying:

```
    // Since this error is really a copy of the
    // "normal" status, it does not also need to be checked
```

But actually `s` might be a different error altogether if it was encountered before we observed a non-ok `IOStatus`

Reply:
Hi Andrew, thanks for the review. The logic is Status S is the overall status, IOStatus io_s only contains the IO related Status. So io_s is a sub-value/partial-value of s. What we need to ensure is 1) Status s bring the earliest error information, 2) any non-ok status in io_s should be stored in s if it is the first error. Therefore, the 
```
if (s.ok()) { 
   s = io_s; 
 } 
```

Appears in multiple locations once io_s has new value.